cmd/manager/main.go

```diff
@@ -115,8 +115,39 @@ func main() {
 	// uniform and structured logs.
 	logf.SetLogger(zap.LoggerTo(os.Stdout))
 
+	if urlOnlyKubeconfig != "" {
+		if err := setKubernetesServiceEnv(urlOnlyKubeconfig); err != nil {
+			log.Error(err, "Terminating")
+			os.Exit(1)
+		}
+	}
+
 	printVersion()
 
+	if sgSetup {
+		log.Info("Setting up AWS Security Groups")
+		cfg, err := config.GetConfig()
+		if err != nil {
+			log.Error(err, "")
+			os.Exit(1)
+		}
+
+		ctx := context.Background()
+
+		client, err := client.New(cfg, client.Options{})
+		if err != nil {
+			log.Error(err, "")
+			os.Exit(1)
+		}
+
+		err = awssgsetup.SetupAWSSecurityGroups(ctx, client)
+		if err != nil {
+			log.Error(err, "")
+			os.Exit(1)
+		}
+		os.Exit(0)
+	}
+
 	// Run the Daemon
 	daemon.Main()
 }
```

Comment:
```suggestion
	} else {
		// Run the Daemon
		daemon.Main()
	}
}
```

Reply:
Why do you think this is better? we often return early out of a function and this feels like the same kind of thing.